### PR TITLE
Add the ability to reply to comments

### PR DESCRIPTION
**Pull request by n-wach**
UI obviously needs some work:

https://screenshot.googleplex.com/oTNmD12jiF5

### CODE REVIEW

File: src/main/webapp/discussion.js [hunk 1]
```diff
@@ -45,11 +48,38 @@ async function loadDiscussion() {
...
+  const preparedComments = prepareComments(comments);
+  for (const comment of preparedComments) {
     ELEMENT_DISCUSSION.appendChild(createComment(comment));
   }
 }
 
+/**
+ * Organizes comments into threads with nested replies.
+ *
+ * <p>All comments are sent from the servlet, the client needs to organize
```
Comment (amuamushu) [inline src/main/webapp/discussion.js:60]:
```suggestion
 * <p>All comments are sent from the servlet; the client needs to organize
```

File: src/main/webapp/view.js [hunk 2]
```diff
@@ -14,7 +14,6 @@
 
 const PARAM_ID = 'id';
```
Comment (amuamushu) [inline src/main/webapp/view.js:16]: Would it be better if these spaces are removed?

File: src/main/webapp/discussion.js [hunk 3]
```diff
@@ -14,29 +14,32 @@
...
 const ELEMENT_DISCUSSION = document.querySelector('#discussion');
 const ELEMENT_POST_TEXTAREA = document.querySelector('#post-textarea');
 
 
-/* exported postAndReload */
 /**
  * Posts comment to {@code ENDPOINT} and reloads the discussion.
  */
-async function postAndReload() {
+async function postAndReload(textarea, parentKey = undefined) {
```
[the remark this answers is not in the document]
Comment (n-wach) [reply src/main/webapp/discussion.js]: Done, also included textarea

File: src/main/webapp/discussion.js (hunk 1 above)
Comment (n-wach) [reply src/main/webapp/discussion.js]: Fixed

File: src/main/webapp/view.js (hunk 2 above)
Comment (n-wach) [reply src/main/webapp/view.js]: Removed

File: src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java [hunk 4]
```diff
@@ -43,17 +43,26 @@
 @WebServlet("/discussion")
 public class DiscussionServlet extends HttpServlet {
   private static final String PARAM_LECTURE = "lecture";
+  private static final String PARAM_PARENT = "parent";
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (jaysaleh) [inline src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java:49]: Decompose this

Comment (jaysaleh) [reply src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java:49]: Same with `doPost()`

**Pull request merged**